docs/src/main/docs/guides/91_mp-tutorial.adoc

```diff
@@ -0,0 +1,1267 @@
+
+import io.helidon.common.CollectionsHelper;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.ws.rs.ApplicationPath;
+import javax.ws.rs.core.Application;
+import java.util.Set;
+
+@ApplicationScoped <1>
+@ApplicationPath("/") <2>
+public class GreetApplication extends Application { <3>
+
+    @Override 
+    public Set<Class<?>> getClasses() { <4>
+        return CollectionsHelper.setOf(GreetResource.class);
+    }
+
+}
+----
+
+
+<1> The class is annotated with `ApplicationScoped` 
+ which specifies that this bean is "application scoped". The application scope
+ is shared between all web service invocations that execute within the same
+ application. 
+<2> The `ApplicationPath` annotation sets the URL path that will be used to 
+ access the application.
+<3> The application class must extend `javax.ws.rs.core.Application`. 
+<4> In the application class, override the 
+ `public Set<Class<?>> getClasses()` method.  This method must return
+ a `Set` of `Class` objects, one for each resource that the application 
+ will serve.  Currently, the application only has one resource, the
+ `GreetResource`.
+
+TIP:  You can learn more about scopes and contexts; and how they are used from the https://javaee.github.io/javaee-spec/javadocs/javax/enterprise/context/package-summary.html[Java EE
+ Specification].
+
+The `GreetResource` is defined in the `GreetResource.java` class as shown
+below: 
+
+[source,java]
+.Initial resource class
+----
+package io.helidon.examples;
+
+import javax.enterprise.context.RequestScoped;
+import javax.json.Json;
+import javax.json.JsonBuilderFactory;
+import javax.json.JsonObject;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+import java.util.Collections;
+
+@Path("/greet") <1>
+@RequestScoped <2>
+public class GreetResource {
+
+    private static final JsonBuilderFactory JSON = Json.createBuilderFactory(Collections.emptyMap());
+
+    @GET
+    @Produces(MediaType.APPLICATION_JSON)
+    public JsonObject getDefaultMessage() { <3>
+        return JSON.createObjectBuilder()
+                .add("message", "Hello World")
+                .build(); <4>
+    }
+
+}
+----
+
+<1> This class is annotated with `Path` which sets the path for this resource
+ as `/greet`.  
+<2> The `RequestScoped` annotation defines that this bean is
+ request scoped.  The request scope is active only for the duration of
+ one web service invocation and it is destroyed at the end of that
+ invocation.
+<3> A `public JsonObject getDefaultMessage()` method is defined
+ which is annotated with `GET`, meaning it will accept the HTTP GET method.
+ It is also annotated with `Produces(MediaType.APPLICATION_JSON)` which
+ declares that this method will return JSON data. 
+<4> The method body creates
+ a JSON object containing a single object named "message" with the content
+ "Hello World".  This method will be expanded and improved
+ later in the tutorial.
```

Comment:
Might be worth pointing out that there's no Helidon or MicroProfile here, i.e. this is just a compliant JAX-RS application and to learn more etc. etc. you can go to various JAX-RS tutorials etc.